src/Microsoft.OData.Core/UriParser/SemanticAst/ODataExpandPath.cs

```diff
@@ -60,23 +60,19 @@ private void ValidatePath()
             {
                 if (segment is TypeSegment)

```

Comment:
Instead of retaining this empty block, would changing the `else` block at the bottom as follows be better:
```csharp
                else if (!(segment is TypeSegment))
                {
                    throw new ODataException(ODataErrorStrings.ODataExpandPath_InvalidExpandPathSegment(segment.GetType().Name));
                }
```